easybuild/tools/github.py

```diff
@@ -958,6 +958,55 @@ def not_eligible(msg):
     return res
 
 
+def close_pr(pr, reason):
+    """
+    Close specified pull request
+    """
+    github_user = build_option('github_user')
+    if github_user is None:
+        raise EasyBuildError("GitHub user must be specified to use --close-pr")
+
+    pr_target_account = build_option('pr_target_account')
+    pr_target_repo = build_option('pr_target_repo')
+
+    pr_url = lambda g: g.repos[pr_target_account][pr_target_repo].pulls[pr]
+    status, pr_data = github_api_get_request(pr_url, github_user)
+    if status != HTTP_STATUS_OK:
+        raise EasyBuildError("Failed to get data for PR #%d from %s/%s (status: %d %s)",
+                             pr, pr_target_account, pr_target_repo, status, pr_data)
+
+    if pr_data['state'] == 'closed':
+        raise EasyBuildError("PR #%d from %s/%s is already closed.", pr, pr_target_account, pr_target_repo)
+
+    msg = "\n%s/%s PR #%s was submitted by %s, " % (pr_target_account, pr_target_repo, pr, pr_data['user']['login'])
+    msg += "you are using GitHub account '%s'\n" % github_user
+    print_msg(msg, prefix=False)
+
+    dry_run = build_option('dry_run') or build_option('extended_dry_run')
+
+    if not reason:
+        reason = "(no reason specified)"
+
+    comment = "@%s, this PR is being closed for the following reason(s): %s.\n" % (pr_data['user']['login'], reason)
+    comment += "Please don't hesitate to reopen this PR or add a comment if you feel this contribution is still relevant.\n"
+    comment += "For more information on our policy w.r.t. closing PRs, see "
+    comment += "https://easybuild.readthedocs.io/en/latest/Contributing.html#why-a-pull-request-may-be-closed-by-a-maintainer"
+    post_comment_in_issue(pr, comment, account=pr_target_account, repo=pr_target_repo, github_user=github_user)
+
+    if dry_run:
+        print_msg("[DRY RUN] Closed %s/%s pull request #%s" % (pr_target_account, pr_target_repo, pr), prefix=False)
+    else:
+        github_token = fetch_github_token(github_user)
+        if github_token is None:
+            raise EasyBuildError("GitHub token for user '%s' must be available to use --close-pr", github_user)        
+        g = RestClient(GITHUB_API_URL, username=github_user, token=github_token)
+        pull_url = g.repos[pr_target_account][pr_target_repo].pulls[pr]
+        body = {'state': 'closed'}
+        status, data = pull_url.post(body=body)
```

Comment:
There's a lot of duplication here too, maybe time for a `pr_action` function that gets PR# + `body` as arguments (and `github_token` as optional argument)?

Reply:
@migueldiascosta Does this suggestion still make sense, or not?

Reply:
@boegel I don't know - each case is slightly different, but I may be missing an obvious generalization

Reply:
Just took a look myself, it's indeed not entirely trivial, so let's leave it as is (for now).